client/js/components/actionbar/action-bar.js

```diff
@@ -1,14 +1,33 @@
 var React = require('react');
+var Link = require('react-router-component').Link;
+var classnames = require('classnames');
 var AddEvent = require('./add-event');
+var AddEmployee = require('./add-employee');
+var AllActions = require('./all-actions');
 
-var ActionBar = React.createClass({
+
+var Actionbar = React.createClass({
   render: function () {
+
+    var Action;
+    switch (this.props.hashRoute) {
+      case '#actionbar/add-event': Action = AddEvent; break;
+      case '#actionbar/add-employee': Action = AddEmployee; break;
+      default: Action = AllActions;
+    }
+
+    var classes = classnames({
+      'action-bar-default': true,
+      'action-bar-open': this.props.actionbarOpen
+    });
+
     return (
-      <div className="action-bar-default" id="action-bar">
-        <AddEvent></AddEvent>
+      <div className={classes} id="action-bar">
+        <Link href={window.location.pathname} className="x-close">x</Link>
```

Comment:
Yea so no matter what your url is, could be "/day/hour/minute/second#actionbar/add-event", the link component will dynamically load with an href equal to just the pathname "/day/hour/minute/second". This fits perfectly with the existing open/close logic and keeps your content at the correct route when you close the actionbar.